### PR TITLE
new BPM documentation 

### PR DESCRIPTION
**Pull request by stasmakarov**
Here is a new documentation for Jmix BPM add-on
Not final yet, but ready for review

### CODE REVIEW

File: content/modules/bpm/partials/nav.adoc [hunk 1]
```diff
@@ -1,25 +1,53 @@
 * xref:bpm:index.adoc[]
-** xref:bpm:getting-started.adoc[]
-** xref:bpm:admin-screens.adoc[]
-*** xref:bpm:modeler.adoc[]
-*** xref:bpm:proc-definitions.adoc[]
-*** xref:bpm:proc-instances.adoc[]
-*** xref:bpm:user-groups.adoc[]
-*** xref:bpm:user-tasks.adoc[]
+** xref:bpm:introduction.adoc[]
+** xref:bpm:whats-new.adoc[]
```
Comment (knstvk) [inline content/modules/bpm/partials/nav.adoc]: Remove this document. What's new page is common for the whole framework and add-ons.

File: content/modules/bpm/pages/index.adoc [hunk 2]
```diff
@@ -1,20 +1,24 @@
 = BPM
 
-include::ROOT:partial$outdated-ui-warning.adoc[]
+BPM add-on provides functionality for modeling and running business processes.
 
-BPM add-on is used for modeling and running business processes. Processes are described using BPMN 2.0 notation. The add-on uses https://flowable.com/open-source/[Flowable^] as a business process engine and https://bpmn.io/[bpmn-js^] rendering toolkit for process modeling.
+[[license]]
```
Comment (knstvk) [inline content/modules/bpm/pages/index.adoc]: Remove this and add the same information about license/trial as in other commercial add-ons, e.g. Maps (in the Installation section).

File: content/modules/bpm/pages/bpm-concepts.adoc [hunk 3]
```diff
@@ -0,0 +1,252 @@
+= Concepts
```
Comment (knstvk) [inline content/modules/bpm/pages/bpm-concepts.adoc]: Rename to "BPM Concepts" to not confuse with the framework's Concepts page

File: content/modules/bpm/pages/introduction.adoc [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+In Studio, the developer has a xref:bpm:modeler-studio.adoc[*Modeler*] for creating BPMN process models, a *Process Form Wizard* (in View Creation Wizard), and a BPM section in the *Jmix Tool Window*.
+
+image::introduction/bpm-components.png[,800]
+
+In the web application, you receive the process xref:bpm:process-modeling.adoc#process-modeling-web[*Modeler*] almost the same as in Studio by functionality, a set of administrative views, and views for regular users allowing they to start processes and work with their tasks.
+
+
+Besides, some extensions become available when you model the process:
+
+* xref:bpm:bpmn/jmix-entity-data-task.adoc[*Entity Data Task*] -- allows to create, modify, and load Jmix Entities directly from the BPMN model.
```
Comment (knstvk) [inline content/modules/bpm/pages/introduction.adoc]: "allows someone to".
See https://youtrack.haulmont.com/articles/PLAT-A-1082/Tips-and-Tricks

File: content/modules/bpm/pages/introduction.adoc [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+[[architecture]]
+== Architecture Overview
+
+*BPM add-on* includes components available in *Studio* and in the *web application* at runtime.
+In Studio, the developer has a xref:bpm:modeler-studio.adoc[*Modeler*] for creating BPMN process models, a *Process Form Wizard* (in View Creation Wizard), and a BPM section in the *Jmix Tool Window*.
+
+image::introduction/bpm-components.png[,800]
+
+In the web application, you receive the process xref:bpm:process-modeling.adoc#process-modeling-web[*Modeler*] almost the same as in Studio by functionality, a set of administrative views, and views for regular users allowing they to start processes and work with their tasks.
```
Comment (knstvk) [inline content/modules/bpm/pages/introduction.adoc]: "allowing them"

File: content/modules/bpm/partials/nav.adoc [hunk 6]
```diff
@@ -1,25 +1,53 @@
...
-** xref:bpm:auto-deployment.adoc[]
-** xref:bpm:api.adoc[]
-//** xref:bpm:sending-notifications.adoc[]
\ No newline at end of file
+** xref:bpm:bpmn/multi-instance-activities.adoc[]
+** xref:bpm:bpmn/transactions.adoc[]
+** xref:bpm:error-handling.adoc[]
+** xref:bpm:jmix-listeners.adoc[]
+** xref:bpm:dmn-1-3.adoc[]
+** Expressions
```
Comment (knstvk) [inline content/modules/bpm/partials/nav.adoc]: Remove not implemented topics

**Review by knstvk — CHANGES_REQUESTED** [review]
Fix also the build warnings:
```
[18:28:20.518] WARN (asciidoctor): section title out of sequence: expected level 1, got level 3
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/bpmn-business-rule-task.adoc:5
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
[18:28:20.521] WARN (asciidoctor): section title out of sequence: expected level 1, got level 3
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/bpmn-business-rule-task.adoc:12
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
[18:28:20.598] WARN (asciidoctor): section title out of sequence: expected level 2, got level 3
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/bpmn-events.adoc:1465
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
[18:28:20.599] WARN (asciidoctor): section title out of sequence: expected level 2, got level 3
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/bpmn-events.adoc:1471
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
[18:28:20.669] WARN (asciidoctor): section title out of sequence: expected level 1, got level 3
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/bpmn-script-task.adoc:15
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
[18:28:20.670] WARN (asciidoctor): section title out of sequence: expected level 1, got level 3
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/bpmn-script-task.adoc:21
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
[18:28:20.690] WARN (asciidoctor): section title out of sequence: expected level 1, got level 3
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/bpmn-service-task.adoc:7
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
[18:28:20.691] WARN (asciidoctor): section title out of sequence: expected level 1, got level 3
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/bpmn-service-task.adoc:13
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
[18:28:20.709] WARN (asciidoctor): skipping reference to missing attribute: myservicebean
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/bpmn-service-task.adoc
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
[18:28:20.711] WARN (asciidoctor): skipping reference to missing attribute: 3
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/bpmn-service-task.adoc
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
[18:28:20.718] WARN (asciidoctor): section title out of sequence: expected level 2, got level 3
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/bpmn-subprocesses.adoc:25
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
[18:28:20.719] WARN (asciidoctor): section title out of sequence: expected level 2, got level 3
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/bpmn-subprocesses.adoc:34
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
[18:28:20.721] WARN (asciidoctor): section title out of sequence: expected level 2, got level 3
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/bpmn-subprocesses.adoc:74
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
[18:28:20.722] WARN (asciidoctor): section title out of sequence: expected level 2, got level 3
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/bpmn-subprocesses.adoc:90
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
[18:28:20.726] WARN (asciidoctor): section title out of sequence: expected level 2, got level 3
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/bpmn-subprocesses.adoc:191
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
[18:28:20.727] WARN (asciidoctor): section title out of sequence: expected level 2, got level 3
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/bpmn-subprocesses.adoc:197
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
[18:28:20.750] WARN (asciidoctor): section title out of sequence: expected level 1, got level 3
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/bpmn-user-task.adoc:17
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
[18:28:20.751] WARN (asciidoctor): section title out of sequence: expected level 1, got level 3
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/bpmn-user-task.adoc:23
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
[18:28:20.775] WARN (asciidoctor): skipping reference to missing attribute: somedatevariable
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/bpmn-user-task.adoc
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
[18:28:20.800] WARN (asciidoctor): section title out of sequence: expected level 1, got level 3
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/jmix-entity-data-task.adoc:11
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
[18:28:20.801] WARN (asciidoctor): section title out of sequence: expected level 1, got level 3
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/jmix-entity-data-task.adoc:24
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
[18:28:20.857] WARN (asciidoctor): section title out of sequence: expected level 1, got level 2
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/multi-instance-activities.adoc:16
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
[18:28:20.858] WARN (asciidoctor): section title out of sequence: expected level 1, got level 2
    file: /Users/knst/work/jmix/docs/jmix-docs-bpm/content/modules/bpm/pages/bpmn/multi-instance-activities.adoc:23
    source: /Users/knst/work/jmix/docs/jmix-docs-bpm (refname: fork/stasmakarov/main <worktree>, start path: content)
```

File: content/modules/bpm/pages/bpm-concepts.adoc (hunk 3 above)
Comment (stasmakarov) [reply content/modules/bpm/pages/bpm-concepts.adoc]: Done

File: content/modules/bpm/partials/nav.adoc (hunk 1 above)
Comment (stasmakarov) [reply content/modules/bpm/partials/nav.adoc]: Done

File: content/modules/bpm/pages/introduction.adoc (hunk 4 above)
Comment (stasmakarov) [reply content/modules/bpm/pages/introduction.adoc]: this block is re-writed.

File: content/modules/bpm/pages/introduction.adoc (hunk 5 above)
Comment (stasmakarov) [reply content/modules/bpm/pages/introduction.adoc]: re-writed

File: content/modules/bpm/pages/index.adoc (hunk 2 above)
Comment (stasmakarov) [reply content/modules/bpm/pages/index.adoc]: Done

File: content/modules/bpm/partials/nav.adoc (hunk 6 above)
Comment (stasmakarov) [reply content/modules/bpm/partials/nav.adoc]: Done

**Review by knstvk — APPROVED** [review]

**Pull request merged**